### PR TITLE
Screens

### PR DESCRIPTION
**Pull request by tanishpvt**
Added On Boarding Screens for enhancement admin can change the assets as per needed, added as refence

### CODE REVIEW

File: sample_client/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,47 +1,33 @@
...
-   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, 
-   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND 
-   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS 
-   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN 
-   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
-   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-   THE SOFTWARE.
-
- -->
```
Comment (Akshay0701) [inline sample_client/src/main/AndroidManifest.xml:24]: i think you shouldn't remove copyright heads from any files :)

File: sample_client/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,47 +1,33 @@
...
-            >
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
+            android:screenOrientation="portrait">
         </activity>
     </application>
+
+    <uses-permission android:name="android.permission.INTERNET" />
```
Comment (Akshay0701) [inline sample_client/src/main/AndroidManifest.xml:31]: is there any strong reason you change location of "INTERNET" ?

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/LoadingScreen.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.owncloud.android.lib.sampleclient;
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/LoadingScreen.java:1]: please add copyright head.

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/LoadingScreen.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+public class LoadingScreen extends AppCompatActivity {
+
+    ProgressBar progressBar;
+    TextView textView;
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_loading_screen);
+//        Objects.requireNonNull(getSupportActionBar()).hide();
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/LoadingScreen.java:22]: revert the comment.

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/MainActivity.java [hunk 5]
```diff
@@ -76,6 +77,9 @@ public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.main);
 
+        getActionBar().hide();
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/MainActivity.java:80]: what is the reason to hide ActionBar in MainActivity ?

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/ProgressBarAnimation.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package com.owncloud.android.lib.sampleclient;
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/ProgressBarAnimation.java:1]: please add copyright head.

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/ProgressBarAnimation.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+public class ProgressBarAnimation extends Animation {
+
+    private Context context;
+    private ProgressBar progressBar;
+    private TextView textView;
+    private Float from;
+    private Float to;
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/ProgressBarAnimation.java:18]:
```suggestion
    private Context mContext;
    private ProgressBar mProgressBar;
    private TextView mTextView;
    private Float mFrom;
    private Float mTo;
```
it would be great if you follow code style(see others files as well)

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/ProgressBarAnimation.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+    @SuppressLint("ResourceAsColor")
+    @Override
+    protected void applyTransformation(float interpolatedTime, Transformation t) {
+        super.applyTransformation(interpolatedTime, t);
+
+        float value=from+(to-from)*interpolatedTime;
+        progressBar.setProgress((int)value);
+        textView.setText((int)value+"%");
+        if(value==to){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/ProgressBarAnimation.java:36]:
```suggestion
        if (value == to) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
+package com.owncloud.android.lib.sampleclient;
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java:1]: copyright**

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+import androidx.viewpager.widget.PagerAdapter;
+import com.airbnb.lottie.LottieAnimationView;
+
+public class SliderAdapterIntro extends PagerAdapter {
+
+    Context context;
+    LayoutInflater layoutInflater;
+
+    public SliderAdapterIntro(Context context){
+        this.context=context;
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java:20]:
```suggestion
        this.context = context;
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    //array of images
+    public String[] slide_images={
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java:23]:
```suggestion
    public String[] slide_images = {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        return view==(RelativeLayout)object;
+    }
+
+    @NonNull
+    @Override
+    public Object instantiateItem(@NonNull ViewGroup container, int position) {
+        layoutInflater=(LayoutInflater)context.getSystemService(context.LAYOUT_INFLATER_SERVICE);
+        View view =layoutInflater.inflate(R.layout.slide_layout,container,false);
+
+        //ImageView slideImageView=(ImageView)view.findViewById(R.id.intro_img);
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java:74]: revert the comment if it's not necessary

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        TextView slideTitle=(TextView) view.findViewById(R.id.intro_title);
+        LottieAnimationView slideImageView=(LottieAnimationView)view.findViewById(R.id.intro_img);
+        //slideImageView.setSpeed(200f);
+
+
+        TextView slideDescription=(TextView) view.findViewById(R.id.intro_description);
+
+        slideImageView.setAnimation(slide_images[position]);
+        slideTitle.setText(slide_title[position]);
+       slideDescription.setText(slide_description[position]);
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/SliderAdapterIntro.java:84]:
```suggestion
        slideDescription.setText(slide_description[position]);
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 14]
```diff
@@ -0,0 +1,247 @@
+package com.owncloud.android.lib.sampleclient;
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:1]: copyright**

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+        mDotsLayout=findViewById(R.id.dontslinear_layout);
+        back=findViewById(R.id.backBtn);
+        next_finish=findViewById(R.id.next_finishBtn);
+
+        sliderAdapterIntro=new SliderAdapterIntro(this);
+
+        viewPager.setAdapter(sliderAdapterIntro);
+        addDotsIndicator(0);
+        viewPager.addOnPageChangeListener(viewListner);
+//        getSupportActionBar().hide();
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:47]: revert the comment*

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 16]
```diff
@@ -0,0 +1,247 @@
...
+
+    @SuppressLint("ResourceType")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_intropage);
+        viewPager=findViewById(R.id.viewpager);
+        mDotsLayout=findViewById(R.id.dontslinear_layout);
+        back=findViewById(R.id.backBtn);
+        next_finish=findViewById(R.id.next_finishBtn);
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:40]:
```suggestion
        viewPager = findViewById(R.id.viewpager);
        mDotsLayout = findViewById(R.id.dontslinear_layout);
        back = findViewById(R.id.backBtn);
        next_finish = findViewById(R.id.next_finishBtn);
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+//        getSupportActionBar().hide();
+
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
+
+
+        //onclick
+        next_finish.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (currentPage==mDots.length-1){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:56]:
```suggestion
                if (currentPage == mDots.length - 1) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+                    .create()
+                    .show();
+            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+            //                                          int[] grantResults)
+            // to handle the case where the user grants the permission. See the documentation
+            // for ActivityCompat#requestPermissions for more details.
+            //return;
+        }
+
+*/
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:159]: is this bloc of code required?

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 19]
```diff
@@ -0,0 +1,247 @@
...
+
+
+
+    public void addDotsIndicator(int position){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:163]:
```suggestion
    public void addDotsIndicator(int position) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 20]
```diff
@@ -0,0 +1,247 @@
...
+        mDots=new TextView[sliderAdapterIntro.getCount()];
+        mDotsLayout.removeAllViews();
+        for (int i=0;i<mDots.length;i++){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:166]:
```suggestion
        for (int i=0 ;i<mDots.length; i++) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 21]
```diff
@@ -0,0 +1,247 @@
...
+            mDots[i]=new TextView(this);
+            mDots[i].setText(Html.fromHtml("&#8226"));
+            mDots[i].setTextSize(25f);
+            mDots[i].setTextColor(getResources().getColor(R.color.colorAccent));
+            mDotsLayout.addView(mDots[i]);
+        }
+        if(mDots.length>0){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:173]:
```suggestion
        if (mDots.length>0) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 22]
```diff
@@ -0,0 +1,247 @@
...
+        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
+        }
+
+        @Override
+        public void onPageSelected(int i) {
+            addDotsIndicator(i);
+            currentPage=i;
+
+            if(i==0){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:189]:
```suggestion
            currentPage = i;

            if (i == 0) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 23]
```diff
@@ -0,0 +1,247 @@
...
+                next_finish.setEnabled(true);
+                back.setEnabled(false);
+                back.setVisibility(View.INVISIBLE);
+
+                next_finish.setText("Next");
+                back.setText("");
+            }
+            else if(i==mDots.length-1){
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:197]:
```suggestion
            else if (i == mDots.length - 1) {
```

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 24]
```diff
@@ -0,0 +1,247 @@
...
+        }
+    };
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+
+
+//        final FirebaseAuth mAuth=FirebaseAuth.getInstance();
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:228]: revert the comment.

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java [hunk 25]
```diff
@@ -0,0 +1,247 @@
...
+        String introShow=prefs.getString("isIntroShow","");
+
+        if(introShow.equals("")) {
+            //  Toast.makeText(this, "", Toast.LENGTH_SHORT).show();
```
Comment (Akshay0701) [inline sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:232]: how are you going to handle it if introShow is empty?

File: sample_client/src/main/res/layout/slide_layout.xml [hunk 26]
```diff
@@ -0,0 +1,83 @@
...
+                android:layout_below="@+id/intro_title"
+                android:layout_centerHorizontal="true"
+                android:layout_gravity="center"
+                android:layout_margin="20dp"
+                android:layout_marginTop="5dp"
+                android:gravity="center"
+                android:padding="10dp"
+                android:text="Description sasdsadsasdsadsad DescriptionDescriptionDescription
+Description Descr iptionDescr iptionDe scriptionDe scriptionDescription
+  Descripti onDescriptionDescriptionDescription Description Description Description"
```
Comment (Akshay0701) [inline sample_client/src/main/res/layout/slide_layout.xml:75]: is this string relevant to the user?

File: sample_client/src/main/res/values/strings.xml [hunk 27]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--  ownCloud Android Library is available under MIT license
+<?xml version="1.0" encoding="utf-8"?><!--  ownCloud Android Library is available under MIT license
```
Comment (Akshay0701) [inline sample_client/src/main/res/values/strings.xml:1]: is there any strong reason to change this?

**Review by Akshay0701 — COMMENTED** [review]
Thank you @tanishpvt for the contribution, 
There are few nits and things which is unnecessarily added in code, 
and it would be really great if you write clean and follow code style (you can refer to other files)

File: sample_client/src/main/AndroidManifest.xml (hunk 1 above)
Comment (tanishpvt) [reply sample_client/src/main/AndroidManifest.xml:24]: not removed

Comment (tanishpvt) [reply sample_client/src/main/AndroidManifest.xml:24]: OKAY, FIXED

File: sample_client/src/main/AndroidManifest.xml (hunk 2 above)
Comment (tanishpvt) [reply sample_client/src/main/AndroidManifest.xml:31]: Fixed

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/LoadingScreen.java (hunk 4 above)
Comment (tanishpvt) [reply sample_client/src/main/java/com/owncloud/android/lib/sampleclient/LoadingScreen.java:22]: done

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java (hunk 24 above)
Comment (tanishpvt) [reply sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:228]: okay

File: sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java (hunk 25 above)
Comment (tanishpvt) [reply sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java:232]: Done with it by providing values

File: sample_client/src/main/res/layout/slide_layout.xml (hunk 26 above)
Comment (tanishpvt) [reply sample_client/src/main/res/layout/slide_layout.xml:75]: No. its jst a dummy

File: sample_client/src/main/res/values/strings.xml (hunk 27 above)
Comment (tanishpvt) [reply sample_client/src/main/res/values/strings.xml:1]: No

**Comment by jesmrec** [comment]
@tanishpvt Could you detail the content of the PR? what is it intended for, what has changed... some screenshots will help to understand as well.

**Comment by abelgardep** [comment]
Thanks for the PR, but the sample client was removed from the library. So, this PR makes no sense anymore.
